tests-e2e/core/gitopsDeployment_test.go

```diff
@@ -0,0 +1,534 @@
+package core
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	managedgitopsv1alpha1 "github.com/redhat-appstudio/managed-gitops/backend/apis/managed-gitops/v1alpha1"
+	fixture "github.com/redhat-appstudio/managed-gitops/tests-e2e/fixture"
+	gitopsDeplFixture "github.com/redhat-appstudio/managed-gitops/tests-e2e/fixture/gitopsdeployment"
+	"github.com/redhat-appstudio/managed-gitops/tests-e2e/fixture/k8s"
+	typed "k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	name    = "my-gitops-depl"
+	repoURL = "https://github.com/redhat-appstudio/gitops-repository-template"
+)
+
+var _ = Describe("GitOpsDeployment E2E tests", func() {
+
+	Context("Create, Update and Delete a GitOpsDeployment ", func() {
+		k8sClient, err := fixture.GetKubeClient()
+
+		Expect(err).To(BeNil())
+		ctx := context.Background()
+
+		It("Should ensure succesfull creation of GitOpsDeployment", func() {
+
+			Expect(fixture.EnsureCleanSlate()).To(Succeed())
+
+			By("creating the GitOpsDeployment")
+
+			gitOpsDeploymentResource := buildGitOpsDeploymentResource(name,
+				repoURL, "environments/overlays/dev",
+				managedgitopsv1alpha1.GitOpsDeploymentSpecType_Automated)
+
+			err := k8s.Create(&gitOpsDeploymentResource)
+			Expect(err).To(Succeed())
+
+			Eventually(gitOpsDeploymentResource, "2m", "1s").Should(
+				SatisfyAll(
+					gitopsDeplFixture.HaveSyncStatusCode(managedgitopsv1alpha1.SyncStatusCodeSynced),
+					gitopsDeplFixture.HaveHealthStatusCode(managedgitopsv1alpha1.HeathStatusCodeHealthy),
+				),
+			)
+			err = k8s.Delete(&gitOpsDeploymentResource)
+			Expect(err).To(Succeed())
+			err = k8s.Get(&gitOpsDeploymentResource)
+			Expect(err).ToNot(Succeed())
+
+		})
+
+		It("Should ensure synchronicity of create and update of GitOpsDeployment", func() {
+			Expect(fixture.EnsureCleanSlate()).To(Succeed())
+
+			By("ensurng no updates are done in existing deployment, if CR is submitted again without any changes")
+
+			gitOpsDeploymentResource := buildGitOpsDeploymentResource(name,
+				repoURL, "environments/overlays/dev",
+				managedgitopsv1alpha1.GitOpsDeploymentSpecType_Automated)
+
+			err := k8s.Create(&gitOpsDeploymentResource)
+			Expect(err).To(Succeed())
+
+			gitOpsDeploymentResource.Spec.Source.RepoURL = repoURL
+			gitOpsDeploymentResource.Spec.Source.Path = "environments/overlays/dev"
+
+			err = k8s.Update(&gitOpsDeploymentResource)
+			Expect(err).To(Succeed())
+
+			err = k8sClient.Get(ctx, client.ObjectKeyFromObject(&gitOpsDeploymentResource), &gitOpsDeploymentResource)
+			Expect(gitOpsDeploymentResource.Spec.Source.RepoURL).To(Equal(repoURL))
+			Expect(gitOpsDeploymentResource.Spec.Source.Path).To(Equal("environments/overlays/dev"))
+
+			Eventually(gitOpsDeploymentResource, "2m", "1s").Should(
```

Comment:
Rather than use an `Eventually` here, we should use a `Consistently`, like this:
```
Consistently(gitOpsDeploymentResource, "20s", "1s").Should(
				SatisfyAll(
					gitopsDeplFixture.HaveSyncStatusCode(managedgitopsv1alpha1.SyncStatusCodeSynced),
					gitopsDeplFixture.HaveHealthStatusCode(managedgitopsv1alpha1.HeathStatusCodeHealthy),
				),
			)
```
(pseudocode)

This means, check every 1 second, for 20 seconds, to make sure that it is still synced and healthy. If it has any other status, even for a second, then the test will fail (which is desired).